include/worker.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2017 Drew Walters <drewwalters96@gmail.com>
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#ifndef WORKER_H
+#define WORKER_H
+
+/* Queue worker function capacity. */
+#define MAX_WORKERS 1
+
+/*
+ * Type definition of worker function added to queue by interrupt handler.
+ */
+typedef void (*work_function)(void *);
+
+/*
+ * Structure containing worker function and its params.
+ */
+struct worker_node {
```

Comment:
"node" implies list (which this is not), so I think something like "work_item" is more appropriate. Feel free to disagree or get creative.